<launch>
  <include file="$(find hrl_pr2_lib)/launch/openni_kinect_polled.launch"/>  
  <node name="visual_servo_node" pkg="visual_servo" type="visual_servo_node" respawn="false" output="screen">
    <remap from="color_image_topic" to="/kinect_head/rgb/image_color"/>
    <remap from="depth_image_topic" to="/kinect_head/depth/image"/>
    <remap from="point_cloud_topic" to="/kinect_head/depth/points"/>

    <param name="tape_hue_value" value="15" type="int"/>
    <param name="tape_hue_threshold" value="50" type="int"/>
    <param name="default_sat_bot_value" value="38" type="int"/>
    <param name="default_sat_top_value" value="220" type="int"/>
    <param name="default_val_value" value="255" type="int"/>

    <!-- 1: inv, 2: pseudo, 3: avg --> 
    <param name="jacobian_type" value="2" type="int"/>
    
    <param name="gain_vel" value="1" type="double"/>
    <param name="gain_rot" value="1" type="double"/>   
  
    <param name="camera_x" value="0" type="double"/>
    <param name="camera_y" value="0.5" type="double"/>
    <param name="camera_z" value="0.0" type="double"/>
    <param name="camera_wx" value="0.4" type="double"/>
    <param name="camera_wy" value="0.5" type="double"/>
    <param name="camera_wz" value="0" type="double"/>
     

    <param name="desired_x" value="0.3" type="double"/>
    <param name="desired_y" value="0.0" type="double"/>
    <param name="desired_z" value="0.8" type="double"/>
    <param name="desired_wx" value="0.3" type="double"/>
    <param name="desired_wy" value="0.1" type="double"/>
    <param name="desired_wz" value="0" type="double"/>
     
    <param name="hand_x" value="-5.0" type="double"/>
    <param name="hand_y" value="-4.5" type="double"/>
    <param name="hand_z" value="4.0" type="double"/>
    <param name="hand_wx" value="0.712" type="double"/>
    <param name="hand_wy" value="-0.412" type="double"/>
    <param name="hand_wz" value="-0.601" type="double"/>
    
    <param name="sim_time" value="8" type="double"/>
    <param name="feature_size" value="3" type="double"/>
    <!-- percentage -->
    <param name="sim_noise_z" value="0" type="double"/>
    
    <param name="display_wait_ms" value="3" type="int"/>
    <param name="workspace_frame" value="/torso_lift_link" type="string"/>

    <param name="crop_min_x" value="20" type="int"/>
    <param name="crop_max_x" value="620" type="int"/>
    <param name="crop_min_y" value="32" type="int"/>
    <param name="crop_max_y" value="480" type="int"/>
    <param name="min_workspace_x" value="-2.0" type="double"/>
    <param name="max_workspace_x" value="1.10" type="double"/>
    <param name="min_workspace_y" value="-1.0" type="double"/>
    <param name="max_workspace_y" value="1.0" type="double"/>
    <param name="min_workspace_z" value="-0.8" type="double"/>
    <param name="max_workspace_z" value="0.6" type="double"/>
    <param name="min_table_z" value="-1.0" type="double"/>
    <param name="max_table_z" value="1.0" type="double"/>
  </node>
</launch>
